fdb-extensions/src/test/java/com/apple/foundationdb/FDBTestBase.java

```diff
@@ -26,9 +26,22 @@
  * Base class that all tests touching FoundationDB should inherit from.
  */
 public abstract class FDBTestBase {
+    private static final int MIN_API_VERSION = 630;
+    private static final int MAX_API_VERSION = 710;
+    private static final String API_VERSION_PROPERTY = "com.apple.foundationdb.apiVersion";
+
+    private static int getAPIVersion() {
+        int apiVersion = Integer.parseInt(System.getProperty(API_VERSION_PROPERTY, "630"));
```

Comment:
Question: Should the default be set to the `FDBFactoryImpl` default (so that when the system default changes, the tests will change automatically)?

Reply:
Unfortunately, not here. The `FDBDatabaseFactory` and `FDBDatabaseFactoryImpl` are in `fdb-record-layer-core`, and this is in `fdb-extensions`, and so the dependencies point in the wrong direction.